### ISSUE TITLE
[Common Lisp] Implement new Concept Exercise: `structures`

### ISSUE THREAD

**Issue by TheLostLambda** [exercism/v3#1799]
This issue describes how to implement the `structures` concept exercise for the Common Lisp track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Learning objectives

- Be able to define a new structure with `defstruct`
- Know which functions are automatically defined by `defstruct` (`make-*`, `copy-*`, `*-p`, and setfable accessors: `*-slot`)
- Be able to construct a structure using `make-*` and keyword arguments
- Be able to check if a value is a certain structure using `*-p`
- Understand how to use the automatically generated accessors (`*-slot`)
- Know how `setf` can be used in combination with the accessors to update a structure
- Be able to assign default values to slots within `defstruct`
- Understand single-inheritance via `:include` works and how slot defaults of the parent can be updated
- Be familiar with common options like `:conc-name` or `:constructor`

## Out of scope

- Classes and what makes them different from structures
- Explaining how `setf` can update "places" – knowledge of `setf` is a prerequisite
- How structures are stored in memory
- The more niche options like `:type`, `:named`, `:print-function`, etc

## Concepts

- `structures`

## Prerequisites

- `assignment`
- `function-definition`
- `variables`
- `symbols`

## Resources to refer to

- [Introduction to Structures](https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node169.html)

### Hints

- [Tutorials Point](https://www.tutorialspoint.com/lisp/lisp_structures.htm)

### After

- [Advanced Options](https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node173.html)

## Representer

Nothing Yet

## Analyzer

It might be useful to check that `defstruct` is actually being used and the functions aren't being manually defined

## Implementing

There are loads of easy examples of structures that can model the real world. Something like a `person` structure might be a good start as it's easy and logical to later define another structure like `programmer` and `:include person` from it.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
